### ISSUE TITLE
[ansible-layer] File mode specification error: (void-variable company-backends)

### ISSUE THREAD

**Issue by raymond-grointel** [syl20bnr/spacemacs#10848]
#### Description :octocat:
`File mode specification error: (void-variable company-backends)`

#### Reproduction guide :beetle:
- Start Emacs
- Open a yaml file (Ansible)

*Observed behaviour:* :eyes: :broken_heart:
`File mode specification error: (void-variable company-backends)`

*Expected behaviour:* :heart: :smile:
No error raised

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 86b4e6f76)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(ansible auto-completion better-defaults clojure colors dash docker emacs-lisp emoji git helm html
         (javascript :variables node-add-modules-path t)
         lua markdown org osx python ranger rust selectric shell shell-scripts spell-checking syntax-checking treemacs typescript version-control yaml)
```
- System configuration features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-variable company-backends)
  (member 'company-ansible company-backends)
  (if (member 'company-ansible company-backends) company-backends (setq company-backends (cons 'company-ansible company-backends)))
  (progn (if (member 'company-ansible company-backends) company-backends (setq company-backends (cons 'company-ansible company-backends))))
  (if (spacemacs//ansible-should-enable\?) (progn (if (member 'company-ansible company-backends) company-backends (setq company-backends (cons 'company-ansible company-backends)))))
  spacemacs/ansible-company-maybe-enable()
  run-hooks(change-major-mode-after-body-hook text-mode-hook yaml-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook text-mode-hook yaml-mode-hook))
  run-mode-hooks(yaml-mode-hook)
  yaml-mode()
  set-auto-mode-0(yaml-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer main.yml> "~/Documents/SRC/gro/devops/roles/utils/aws_es/tas
```

**Comment by syl20bnr** [comment]
I'll try to reproduce it but I use it everyday without issues.

**Comment by Miciah** [comment]
I was not able to reproduce the issue.  However, #10445 deals with the same issue for other packages.  Perhaps someone who is able to reproduce this issue could try the solution in #10445, adapted for the ansible layer.

**Comment by duianto** [comment]
Confirmed, the error message:
>File mode specification error: (void-variable company-backends)

appears when this example file is opened.
`main.yml`
```yaml
- name: create user groups
  group:
    name: "{{ item }}"
  loop:
    - postgresql
    - nginx-test
    - admin
    - dbadmin
    - hadoop
```

#### System info
<details><summary>
- Ubuntu 18.04
</summary>
<pre>
#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. c5c63f48c)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(ansible auto-completion emacs-lisp git helm markdown multiple-cursors org spell-checking syntax-checking treemacs version-control yaml)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD PDUMPER LCMS2 GMP
</pre>
</details>

<details><summary>
- Windows 1903
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. c5c63f48c)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(ansible autohotkey auto-completion
         (dart :variables dart-server-sdk-path "C:/tools/dart-sdk/" dart-server-enable-analysis-server t)
         emacs-lisp git helm html lsp multiple-cursors org spell-checking syntax-checking neotree version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2
</pre>
</details>

Opened a PR after following Miciah's suggestion:
[ansible] Fix void-variable company-backends #12676

**Comment by duianto** [comment]
The PR has been merged.
[ansible] Fix void-variable company-backends #12676
